src/jit/gentree.cpp

```diff
@@ -10166,6 +10166,18 @@ GenTreePtr                  Compiler::gtFoldExprConst(GenTreePtr tree)
                             goto LNG_OVF;
                         }
                         if ((ltemp/lval2) != lval1) goto LNG_OVF;
+#ifdef UNIX_AMD64_ABI
+                        // There is a clang 3.5 optimizer bug that in case of lval1 and lval2 equal
+                        // of 0x80000000 (MIN_LONG) the above expression (ltemp / lval2) results
+                        // in 0x80000000 (MIN_LONG. This causes an overflowing mul expression 
+                        // to be repalced with a 0, instead of throwing an overflow exception.
```

Comment:
I am not in favor of using any compiler specific of to detect overflow conditions here.

The code that is already there is already correct and is compiler independednt, we don't need to use any compiler specific extensions here as this is not a important area to make changes that are compiler specific.

In general we should only make compiler specific changes in areas that can give us increase throughput.

Since this is a LLVM/Clang compiler bug we should just add some temporary code to handle it until we move to an compiler that has a fix for this issue.